File: datasets/py-keep-code_davinci_001_temp_0.2/HumanEval_14_all_prefixes.py
from typing import List

def all_prefixes(string: str) -> List[str]:
    """ Return list of all prefixes from shortest to longest of the input string
    >>> all_prefixes('abc')
    ['a', 'ab', 'abc']
    """
    if not string:
        return []
    return [string[:i] for i in range(1, len(string) + 1)]


def check(candidate):
    assert candidate('') == []
    assert candidate('asdfgh') == ['a', 'as', 'asd', 'asdf', 'asdfg', 'asdfgh']
    assert candidate('WWW') == ['W', 'WW', 'WWW']

def test_check():
    check(all_prefixes)

test_check()

File: datasets/ts-keep-code_davinci_001_temp_0.2/HumanEval_14_all_prefixes.ts
//Return list of all prefixes from shortest to longest of the input string
// >>> all_prefixes('abc')
// ['a', 'ab', 'abc']
function all_prefixes(string: string): string[] {
    let prefixes: string[] = [];
    for (let i = 0; i < string.length; i++) {
        prefixes.push(string.substring(0, i + 1));
    }
    return prefixes;
}


declare var require: any;
const assert = require('node:assert');


function test() {
  let candidate = all_prefixes;
  assert.deepEqual(candidate(""),[]);
  assert.deepEqual(candidate("asdfgh"),["a", "as", "asd", "asdf", "asdfg", "asdfgh"]);
  assert.deepEqual(candidate("WWW"),["W", "WW", "WWW"]);
}

test();